extensions/wikia/Staging/StagingHooks.class.php

```diff
@@ -26,10 +26,11 @@ static public function onMakeGlobalVariablesScript( &$vars ) {
 	static public function onBeforePageRedirect( $out, &$redirect, &$code ) {
 		if ( !empty( $_SERVER['HTTP_X_STAGING'] ) ) {
 			$stagingEnvName = $_SERVER['HTTP_X_STAGING'];
-			$stagingUrlPart = '://' . $stagingEnvName . '.';
+			$parts = parse_url( $redirect );
 
-			if ( strpos( $redirect, 'wikia.com' ) !== false && strpos( $redirect, $stagingUrlPart ) === false ) {
-				$redirect = str_replace( '://', $stagingUrlPart, $redirect );
+			if ( strpos( $parts['host'], '.wikia.com' ) !== false && strpos( $parts['host'], $stagingEnvName . '.wikia.com' ) === false ) {
+				$url['host'] = str_replace( '.wikia.com', '.' . $stagingEnvName . '.wikia.com', $parts['host'] );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "$url" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

[project-id]: wikia:app